tests/unit-tests/api/utilities/beansPathToUrl.php

```diff
@@ -0,0 +1,338 @@
+<?php
+
+namespace Beans\Framework\Tests\UnitTests\API\Utilities;
+
+use Beans\Framework\Tests\UnitTests\Test_Case;
+use Brain\Monkey\Functions;
+
+/**
+ * Class Tests_Beans_Get
+ *
+ * @package Beans\Framework\Tests\API\Utilities
+ * @group   unit-tests
+ * @group   api
+ */
+class Tests_Beans_Path_To_Url extends Test_Case {
+
+	/**
+	 * Setup test fixture.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once BEANS_TESTS_LIB_DIR . '/api/utilities/functions.php';
+	}
+
+	/**
+	 * Test beans_path_to_url() should bail out when URL.
+	 */
+	public function test_should_bail_out_when_url() {
+		$this->assertSame( 'http://getbeans.io', beans_path_to_url( 'http://getbeans.io' ) );
+		$this->assertSame( 'http://www.getbeans.io', beans_path_to_url( 'http://www.getbeans.io' ) );
+		$this->assertSame( 'https://getbeans.io', beans_path_to_url( 'https://getbeans.io' ) );
+		$this->assertSame( 'https://www.getbeans.io', beans_path_to_url( 'https://www.getbeans.io' ) );
+	}
+
+	/**
+	 * Test beans_path_to_url() should bail out when relative URL.
+	 */
+	public function test_should_bail_out_when_relative_url() {
+		$this->assertSame( '//getbeans.io', beans_path_to_url( '//getbeans.io' ) );
+		$this->assertSame( '//www.getbeans.io', beans_path_to_url( '//www.getbeans.io' ) );
+
+//		$this->assertSame( 'getbeans.io', beans_path_to_url( 'getbeans.io' ) );
```

Comment:
Why are these commented out?

Reply:
These 2 tests fail.  I was thinking about whether it needs to pass for these or not.  I left them in to remind me.  

Why do they fail?  We're using a `preg_match()` and looking for `data`, `//`, `http`, `https`, to let us know that this is a URL.  A relative URL without any of those is not caught.  Instead, it thinks it's a filesystem path.

In another PR, I'll be submitting some refactored code that moves redundant checks into private functions:

- checking for a tilde
- checking for subdirectory
- checking if it's a URL or a filesystem path
- etc.

During that PR, I'll address these 2 lines of code.  If I remember them, they may slip through the cracks.